packages/polymorphic/src/Example/types.spec.tsx

```diff
@@ -0,0 +1,286 @@
+        some content
+      </AdvancedPolymorphic>
+
+      {/* @ts-expect-error href is not allowed on explicit div */}
+      <AdvancedPolymorphic as="div" href="mongodb.design" />
+      {/* @ts-expect-error target is not allowed on explicit div */}
+      <AdvancedPolymorphic as="div" targe="_blank" />
+
+      <AdvancedPolymorphic as="input" />
+
+      <AdvancedPolymorphic as={WrapperWithRef} />
+      <AdvancedPolymorphic as={WrapperWithRef} ref={spanRef} />
+      <AdvancedPolymorphic as={WrapperWithRef} ref={spanRef} darkMode={true} />
+      {/* TODO: ts-expect-error - Must pass the correct ref type */}
+      <AdvancedPolymorphic as={WrapperWithRef} ref={divRef} />
+      {/* @ts-expect-error - Theme is not a prop on WrapperWithRef */}
+      <AdvancedPolymorphic as={WrapperWithRef} ref={spanRef} theme={'dark'} />
+      {/* @ts-expect-error - href is not a prop on WrapperWithRef */}
+      <AdvancedPolymorphic as={WrapperWithRef} ref={spanRef} href=".design" />
+    </>;
+  });
+
+  test.skip('AdvancedPolymorphicWithRef', () => {
+    <>
+      <AdvancedPolymorphicWithRef />
+      <AdvancedPolymorphicWithRef>some content</AdvancedPolymorphicWithRef>
+      <AdvancedPolymorphicWithRef as="div" />
+      <AdvancedPolymorphicWithRef as="div" ref={divRef} />
+      {/* @ts-expect-error - Must pass the correct ref type */}
+      <AdvancedPolymorphicWithRef as="div" ref={anchorRef} />
+      <AdvancedPolymorphicWithRef as="div" ref={divRef}>content</AdvancedPolymorphicWithRef>
+      <AdvancedPolymorphicWithRef key="some-key" />
+
+      {/* @ts-expect-error - Require href when as="a" */}
+      <AdvancedPolymorphicWithRef as="a" />
+      <AdvancedPolymorphicWithRef as="a" href="mongodb.design" />
+      <AdvancedPolymorphicWithRef as="a" href="mongodb.design" ref={anchorRef} />
+      <AdvancedPolymorphicWithRef as="a" href="mongodb.design">content</AdvancedPolymorphicWithRef>
+
+      {/* @ts-expect-error href is not allowed on explicit div */}
+      <AdvancedPolymorphicWithRef as="div" href="mongodb.design" />
+      {/* @ts-expect-error target is not allowed on explicit div */}
+      <AdvancedPolymorphicWithRef as="div" target="_blank" />
+
+      <AdvancedPolymorphicWithRef as="input" />
+
+      <AdvancedPolymorphicWithRef as={WrapperWithRef} />
+      <AdvancedPolymorphicWithRef as={WrapperWithRef} ref={spanRef} />
+      <AdvancedPolymorphicWithRef as={WrapperWithRef} ref={spanRef} darkMode={true} />
+      {/* TODO: ts-expect-error - Must pass the correct ref type */}
+      <AdvancedPolymorphicWithRef as={WrapperWithRef} ref={divRef} />
+      {/* @ts-expect-error - Theme is not a prop on WrapperWithRef */}
+      <AdvancedPolymorphicWithRef as={WrapperWithRef} ref={spanRef} theme={'dark'} />
+      {/* @ts-expect-error - href is not a prop on WrapperWithRef */}
+      <AdvancedPolymorphicWithRef as={WrapperWithRef} ref={spanRef} href=".design" />
+    </>;
+  });
+
+  test.skip('ExampleInferred', () => {
+    <>
```

Comment:
Can we add a test that shows that we support NextLink components or should this be done on a per-component basis?

Reply:
I can add a test here

Reply:
Apparently `NextLink` doesn't spread `rest`? 

This TSX:
```tsx
<NextLink href="mongodb.design" data-testid="next" target="_blank">
  Content
</NextLink>,
```

Renders to: 
```html
<a href="/mongodb.design">
  Content
</a>
```